### PR TITLE
Project plan update: Sukanta Roy

### PR DESCRIPTION
**Pull request by sukanta-27**
- [x] Added week 3-4 progress report
- [x] Updated project plan to accordingly and also included https://github.com/TEAMMATES/teammates/issues/8894 and https://github.com/TEAMMATES/teammates/issues/8895 in the project plan as suggested by Prof. Damith.

@xpdavid @bqnguyen94 Ready for review.

### CODE REVIEW

File: Sukanta_Roy/progress.md [hunk 1]
```diff
@@ -47,30 +47,49 @@ Week 2 Retrospect:
...
 	- [ ] Add necessary JS tests
 	- [ ] Add frontend UI tests
 
+Week 3 Work:
+- [x] Merged: [Instructor: Edit session: Rubric question: Add front-end validation for empty weights](https://github.com/TEAMMATES/teammates/issues/8631)
+- [x] Merged: [Rubric question add/edit action test: Add missing parameter and additional tests](https://github.com/TEAMMATES/teammates/issues/8855)
+- [x] Finished adding backend implementation for MCQ weights feature in [Instructor: MCQ questions (single answer): support attaching weights to options #2776](https://github.com/TEAMMATES/teammates/pull/8860)
+
+Week 3 Retrospect:
+In this week, created the pull request for MCQ weights and added backend for MCQ weights. It took longer than expected to add tests for backend as I am relatively new to Unit testing. Apart from that, no major setbacks.
```
Comment (xpdavid) [inline Sukanta_Roy/progress.md:63]: I believe you have learnt a lot about how to write unit test and you are doing good actually :)

File: Sukanta_Roy/progress.md [hunk 2]
```diff
@@ -47,30 +47,49 @@ Week 2 Retrospect:
...
+- [x] Code review and fix
+
+Week 4 Work:
+- [x] Added new UI for MCQ weights by updating question templates.
+- [x] Attached JS functionalities to it
+- [x] Added front-end UI tests to MCQ weights
+- [x] Added necessary End-to-End tests for MCQ weights
+
+Week 4 Retrospect:
+In this week, I worked on front-end part of MCQ weights and added necessary tests to it. Could not complete the calculation of statistics part within this week as working on JS functionalities and the tests took longer than what I had initially anticipated.
```
Comment (xpdavid) [inline Sukanta_Roy/progress.md:78]: I how the experience will help you implement MSQ feature

**Review by xpdavid — APPROVED** [review]

File: Sukanta_Roy/progress.md (hunk 1 above)
Comment (sukanta-27) [reply Sukanta_Roy/progress.md:63]: @xpdavid Thank you! Yes, I have learnt a lot about testing while working on MCQ questions. Implementing weights feature for MSQ will be a lot easier.

**Comment by xpdavid** [comment]
@bqnguyen94 Merge the PR after your approval :)

**Review by bqnguyen94 — APPROVED** [review]

**Pull request merged**